### PR TITLE
docs: Fix #997

### LINKED ISSUE #997: [docs] Fix the document as to cache on leo_gateway
As described on comments of https://github.com/leo-project/leofs/issues/936, cache related documents should have been fixed as below.

- Cache is only available to small objects when cache_disc_capacity = 0.
    - IOW, An object which size are larger than cache_disc_threshold_len never be cached if cache_disc_capacity = 0.

- There are two different cache mechanisms for each (small and large objects).
  - For small objects
    - Cache happens in both cases PUT(write through) and GET(read through)
    - Stored to memory if the size of an object < cache.cache_disc_threshold_len
    - Stored to DISK if the size of an object >= cache.cache_disc_threshold_len
  - For large objects
    - Cache happens ONLY in case GET(read through)
    - Stored to always DISK ONLY if disk cache is enabled (cache_disc_capacity > 0)
    - the whole part object is written as a file on the host file system during processing a GET request (a chunk level cache is not implemented)

### PR DESCRIPTION
**Pull request by mocchira**
Preview is available on https://mocchira.github.io/leofs/admin/settings/leo_gateway/#configurations-which-may-affect-the-cache-behavior

### CODE REVIEW

File: docs/admin/settings/leo_gateway.md [hunk 1]
```diff
@@ -131,6 +131,22 @@ And also, when size of a requested object more than `cache.cache_max_content_len
 
 ![Figure: Disk Cache Limits](../../assets/leofs-gateway-disk-cache-size.png)
 
+#### Configurations which may affect the cache behavior
+Whether or not an object is cached on LeoGateway is determined by below logics.
+
+- LeoGateway's cache feature is only available to small objects when cache.cache_disc_capacity = 0.
```
Comment (yosukehara) [inline docs/admin/settings/leo_gateway.md]: LeoGateway's cache feature is only available to small objects when `cache.cache_disc_capacity = 0`.

File: docs/admin/settings/leo_gateway.md [hunk 2]
```diff
@@ -131,6 +131,22 @@ And also, when size of a requested object more than `cache.cache_max_content_len
...
+- An object which size is larger than cache.cache_disc_threshold_len never be cached if cache.cache_disc_capacity = 0.
```
Comment (yosukehara) [inline docs/admin/settings/leo_gateway.md]: An object which size is larger than `cache.cache_disc_threshold_len` never be cached if `cache.cache_disc_capacity = 0`.

File: docs/admin/settings/leo_gateway.md [hunk 3]
```diff
@@ -131,6 +131,22 @@ And also, when size of a requested object more than `cache.cache_max_content_len
...
+
+How LeoGateway's cache feature works with cache related configurations is described below.
+
+- For small objects
+    - Cache happens in both cases handling a PUT request in the write through way and handling a GET in the read through way
+    - Being stored into memory if the size of an object < cache.cache_disc_threshold_len
```
Comment (yosukehara) [inline docs/admin/settings/leo_gateway.md]: Being stored into memory if the size of an object < `cache.cache_disc_threshold_len`

File: docs/admin/settings/leo_gateway.md [hunk 4]
```diff
@@ -131,6 +131,22 @@ And also, when size of a requested object more than `cache.cache_max_content_len
...
+    - Being stored into Disk if the size of an object >= cache.cache_disc_threshold_len
```
Comment (yosukehara) [inline docs/admin/settings/leo_gateway.md]: Being stored into Disk if the size of an object >= `cache.cache_disc_threshold_len`

File: docs/admin/settings/leo_gateway.md [hunk 5]
```diff
@@ -131,6 +131,22 @@ And also, when size of a requested object more than `cache.cache_max_content_len
...
+- For large objects
+    - Cache happens ONLY in case handling a GET request in the read through way
+    - Being stored into Disk ONLY if disk cache is enabled (cache_disc_capacity > 0)
```
Comment (yosukehara) [inline docs/admin/settings/leo_gateway.md]: Being stored into Disk ONLY if disk cache is enabled (`cache_disc_capacity` > 0)

**Comment by mocchira** [comment]
@yosukehara PTAL.

**Pull request merged**